provider/resources.go

```diff
@@ -4666,6 +4668,8 @@ compatibility shim in favor of the new "name" field.`)
 			"aws_ec2_managed_prefix_list":          {Tok: awsDataSource(ec2Mod, "getManagedPrefixList")},
 			"aws_ec2_transit_gateway_route_tables": {Tok: awsDataSource(ec2Mod, "getTransitGatewayRouteTables")},
 			"aws_ec2_instance_types":               {Tok: awsDataSource(ec2Mod, "getInstanceTypes")},
+			"aws_vpc_ipam":                         {Tok: awsDataSource(ec2Mod, "getVpcIpam")},
```

Comment:
This is `aws_vpc_*`, but everything else we've been putting in `aws_ec2` so just following that.

Reply:
Which module does it show up in the TF documentation site? Alongside which resources? Can we match that for minimum surprise?

Reply:
Well the problem is that we haven't been matching TF so far. The TF docs site has a separate section for `Vpc` (which includes a lot more resources that our Vpc docs section) and a separate `Vpc IPAM` section.

It seems like we've already diverged pretty far from what the TF docs do, but that also might be because they are able to re-organize the resources on the docs site when they want to.

![Screenshot 2025-02-05 at 10 15 05 AM](https://github.com/user-attachments/assets/5b0c5c46-1489-48f2-b77c-5c8a4ed1569e)


Also, it might be a moot point because I didn't realize we already had a `aws_vpc_ipam` `resource` that we put under the `ec2` module so I think we need to also put the datasource there as well.

Reply:
That's a good idea, we should think about how to reorganize these resources for V7. I'll create an issue to track.

Reply:
https://github.com/pulumi/pulumi-aws/issues/5168